### PR TITLE
SCB-730 Add Feign transport for supporting Spring Cloud

### PR DESCRIPTION
**Pull request by Lionelpang**
满足spring cloud下服务间使用openfeign进行服务间通讯使用saga实现分布式事务。实现feign接口调用过程中传递全局事ID和本地事务ID传递。

奉献绵薄之力，经过测试，目前已经可以完整对接feign。
---

### CODE REVIEW

File: omega/omega-transport/omega-transport-feign/src/main/java/org/apache/servicecomb/saga/omega/transport/feign/FeignAutoConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package org.apache.servicecomb.saga.omega.transport.feign;
```
Comment (WillemJiang) [inline omega/omega-transport/omega-transport-feign/src/main/java/org/apache/servicecomb/saga/omega/transport/feign/FeignAutoConfiguration.java]: Please add ASF License here.

**Comment by WillemJiang** [comment]
@Lionelpang Thanks for the contribution.  The patch looks good. I just created a [JIRA](https://issues.apache.org/jira/browse/SCB-730) to track this issue. 
As we need to add the ASF header license to the source file, please submit another PR for it. You can find the license header information by check other java file in the repo.

File: omega/omega-transport/omega-transport-feign/src/main/java/org/apache/servicecomb/saga/omega/transport/feign/FeignAutoConfiguration.java (hunk 1 above)
Comment (Lionelpang) [reply omega/omega-transport/omega-transport-feign/src/main/java/org/apache/servicecomb/saga/omega/transport/feign/FeignAutoConfiguration.java]: Already added

File: omega/omega-transport/omega-transport-feign/src/test/java/org/apache/servicecomb/saga/omega/transport/feign/FeignAutoConfigurationTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
+package org.apache.servicecomb.saga.omega.transport.feign;
```
Comment (WillemJiang) [inline omega/omega-transport/omega-transport-feign/src/test/java/org/apache/servicecomb/saga/omega/transport/feign/FeignAutoConfigurationTest.java]: We need the license header here.

File: omega/omega-transport/omega-transport-feign/src/main/resources/META-INF/spring.factories [hunk 3]
```diff
@@ -0,0 +1,2 @@
+org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
```
Comment (WillemJiang) [inline omega/omega-transport/omega-transport-feign/src/main/resources/META-INF/spring.factories]: We need the license header here.

File: omega/omega-transport/omega-transport-feign/pom.xml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            <artifactId>spring-boot-autoconfigure</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>javax.servlet-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-test</artifactId>
```
Comment (WillemJiang) [inline omega/omega-transport/omega-transport-feign/pom.xml]: I think the scope should be test.

**Comment by WillemJiang** [comment]
@Lionelpang  I just found there some minor issues need to fix. Could you submit another commit to fix those issues?

File: omega/omega-transport/omega-transport-feign/src/main/resources/META-INF/spring.factories (hunk 3 above)
Comment (Lionelpang) [reply omega/omega-transport/omega-transport-feign/src/main/resources/META-INF/spring.factories]: Already added

File: omega/omega-transport/omega-transport-feign/pom.xml (hunk 4 above)
Comment (Lionelpang) [reply omega/omega-transport/omega-transport-feign/pom.xml]: Already added

**Comment by WillemJiang** [comment]
Merge the patch into master branch.  We may need to updated the License file and Distribution file for new added module. @Lionelpang  Please ping me at the [gitter](https://gitter.im/ServiceCombUsers/Lobby). I will walk you through it.

**Pull request merged**